text/0013-trits-t5b1/0013-trits-t5b1.md

```diff
@@ -0,0 +1,322 @@
+    [ 0, 1, 0, 0, -1 ],    [ 1, 1, 0, 0, -1 ],    [ -1, -1, 1, 0, -1 ],   [ 0, -1, 1, 0, -1 ],   [ 1, -1, 1, 0, -1 ],
+    [ -1, 0, 1, 0, -1 ],   [ 0, 0, 1, 0, -1 ],    [ 1, 0, 1, 0, -1 ],     [ -1, 1, 1, 0, -1 ],   [ 0, 1, 1, 0, -1 ],
+    [ 1, 1, 1, 0, -1 ],    [ -1, -1, -1, 1, -1 ], [ 0, -1, -1, 1, -1 ],   [ 1, -1, -1, 1, -1 ],  [ -1, 0, -1, 1, -1 ],
+    [ 0, 0, -1, 1, -1 ],   [ 1, 0, -1, 1, -1 ],   [ -1, 1, -1, 1, -1 ],   [ 0, 1, -1, 1, -1 ],   [ 1, 1, -1, 1, -1 ],
+    [ -1, -1, 0, 1, -1 ],  [ 0, -1, 0, 1, -1 ],   [ 1, -1, 0, 1, -1 ],    [ -1, 0, 0, 1, -1 ],   [ 0, 0, 0, 1, -1 ],
+    [ 1, 0, 0, 1, -1 ],    [ -1, 1, 0, 1, -1 ],   [ 0, 1, 0, 1, -1 ],     [ 1, 1, 0, 1, -1 ],    [ -1, -1, 1, 1, -1 ],
+    [ 0, -1, 1, 1, -1 ],   [ 1, -1, 1, 1, -1 ],   [ -1, 0, 1, 1, -1 ],    [ 0, 0, 1, 1, -1 ],    [ 1, 0, 1, 1, -1 ],
+    [ -1, 1, 1, 1, -1 ],   [ 0, 1, 1, 1, -1 ],    [ 1, 1, 1, 1, -1 ],     [ -1, -1, -1, -1, 0 ], [ 0, -1, -1, -1, 0 ],
+    [ 1, -1, -1, -1, 0 ],  [ -1, 0, -1, -1, 0 ],  [ 0, 0, -1, -1, 0 ],    [ 1, 0, -1, -1, 0 ],   [ -1, 1, -1, -1, 0 ],
+    [ 0, 1, -1, -1, 0 ],   [ 1, 1, -1, -1, 0 ],   [ -1, -1, 0, -1, 0 ],   [ 0, -1, 0, -1, 0 ],   [ 1, -1, 0, -1, 0 ],
+    [ -1, 0, 0, -1, 0 ],   [ 0, 0, 0, -1, 0 ],    [ 1, 0, 0, -1, 0 ],     [ -1, 1, 0, -1, 0 ],   [ 0, 1, 0, -1, 0 ],
+    [ 1, 1, 0, -1, 0 ],    [ -1, -1, 1, -1, 0 ],  [ 0, -1, 1, -1, 0 ],    [ 1, -1, 1, -1, 0 ],   [ -1, 0, 1, -1, 0 ],
+    [ 0, 0, 1, -1, 0 ],    [ 1, 0, 1, -1, 0 ],    [ -1, 1, 1, -1, 0 ],    [ 0, 1, 1, -1, 0 ],    [ 1, 1, 1, -1, 0 ],
+    [ -1, -1, -1, 0, 0 ],  [ 0, -1, -1, 0, 0 ],   [ 1, -1, -1, 0, 0 ],    [ -1, 0, -1, 0, 0 ],   [ 0, 0, -1, 0, 0 ],
+    [ 1, 0, -1, 0, 0 ],    [ -1, 1, -1, 0, 0 ],   [ 0, 1, -1, 0, 0 ],     [ 1, 1, -1, 0, 0 ],    [ -1, -1, 0, 0, 0 ],
+    [ 0, -1, 0, 0, 0 ],    [ 1, -1, 0, 0, 0 ],    [ -1, 0, 0, 0, 0 ]
+]
+```
+
+Decoding a `signed byte` into `5` `balanced` `trits` with the help of the `LUT` is fairly simple:
+- `byte >= 0`: use `byte` as an index for the `LUT` e.g. if the `byte = 42` then `LUT[42] = [ 0, -1, -1, -1, 1 ]` and `0 * 3^0 + (-1) * 3^1 + (-1) * 3^2 + (-1) * 3^3 + 1 * 3^4 = 42`;
+- `byte < 0`: use `byte + 243` as an index for the `LUT` e.g. if the `byte = -42` then `LUT[201] = [ 0, 1, 1, 1, -1 ]` and `0 * 3^0 + 1 * 3^1 + 1 * 3^2 + 1 * 3^3 + (-1) * 3^4 = -42`;
+
+Even though this RFC won't cover every little implementation details, it can be noted that the `LUT` could cover all `256` possible values of a `byte` to avoid invalid accesses. To the missing `byte` values we can simply assign some default mapping to `5` `trits`, eg. `[ 0, 0, 0, 0, 0 ]`. This way runtime checks can be avoided as branching on each iteration of a loop can degrade performance in case of false branch prediction.
+
+Additionally, there could be two variants of the decoding function:
+- checked & slow;
+- unchecked & fast;
+
+The overhead of the `13` additional values is only +~5% to the size of `LUT`.
+
+### Without LUT
```

Comment:
taking into account functions that get/set one trit in this encoding (or any other) without LUT 
it would take nasty bitwise operations, we should think about it if we even want to allow anything without LUT,
IMO, this is a really tiny overhead  considering the Ledger itself, so, i struggle to find why it should be only optional

Reply:
The size of the LUTs is non negligible for some very limited devices. Also, having two different implementations for testing purposes is great.